packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1409,3 +1409,77 @@ class SliverConstrainedCrossAxis extends SingleChildRenderObjectWidget {
     renderObject.maxExtent = maxExtent;
   }
 }
+
+/// Set a flex factor for allocating space in the cross axis direction.
+///
+/// This is a [ParentDataWidget] to be used in [SliverCrossAxisGroup].
+/// The remaining space (cross axis extent of the parent minus the extent of
+/// children with constrained cross axis) will be divided according to the flex values.
+class SliverExpanded extends ParentDataWidget<SliverPhysicalContainerParentData> {
+  /// Creates an object that assigns a [flex] value to the child sliver.
+  ///
+  /// The provided [flex] value must be at least 0.
+  const SliverExpanded({
+    super.key,
+    required this.flex,
+    required Widget sliver,
+  }): assert(flex > 0 && flex < double.infinity),
+      super(child: sliver);
+
+  /// Flex value for allocating cross axis extent left after laying out the children with
+  /// constrained cross axis. The children with flex values will have the remaining extent
+  /// allocated proportionally to their flex value. This must an integer between
+  /// 0 and infinity, exclusive.
+  final int flex;
+
+  @override
+  void applyParentData(RenderObject renderObject) {
+    assert(renderObject.parentData is SliverPhysicalContainerParentData);
+    final SliverPhysicalContainerParentData parentData = renderObject.parentData! as SliverPhysicalContainerParentData;
+    bool needsLayout = false;
+
+    if (parentData.flex != flex) {
+      parentData.flex = flex;
+      needsLayout = true;
+    }
+
+    if (needsLayout) {
+      final AbstractNode? targetParent = renderObject.parent;
+      if (targetParent is RenderObject) {
+        targetParent.markNeedsLayout();
+      }
+    }
+  }
+
+  @override
+  Type get debugTypicalAncestorWidgetClass => SliverCrossAxisGroup;
+}
+
+
+/// A sliver that places multiple sliver children in a linear array along
+/// the cross axis.
+///
+/// Each child has either a flex or extent value associated with them.
+/// These flex/extent values are assigned with the [SliverExpanded] or
+/// [SliverConstrainedCrossAxis] [ParentDataWidget], both of which takes
+/// in their respective value and a sliver child. If any other sliver
+/// is provided as a child, it will by default be assigned a flex value of 1.
+///
+/// See also:
+/// * [SliverExpanded], which is the [ParentDataWidget] for setting a flex
+///   value to a widget.
+/// * [SliverConstrainedCrossAxis], which is the [ParentDataWidget] for setting
+///   a extent to constrain the widget to.
+class SliverCrossAxisGroup extends MultiChildRenderObjectWidget {
```

Comment:
Added.

Reply:
Then why is it being overridden here? Should this extend the RenderProxySliver class instead of RenderSliver?
It not being tested somewhere else is not a reason to not test it here.

Reply:
Oh I think github got confused, I meant to leave that comment on a different one.